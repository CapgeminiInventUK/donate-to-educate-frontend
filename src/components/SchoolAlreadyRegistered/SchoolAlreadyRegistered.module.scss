@use '@assets/global.scss' as g;

.container {
  background-color: g.$help-grey;
  padding: 40px;
  color: g.$primary-dark-blue;
  margin-bottom: 40px;

  h2 {
    margin-top: 0;
  }

  a {
    color: g.$primary-dark-blue;
  }

  .link {
    text-decoration: underline;
    color: g.$link-blue;
    font-size: 18px;

    @include g.sm-breakpoint {
      font-size: 14px;
    }
  }

  .homeIcon {
    color: g.$link-blue;
  }
}
