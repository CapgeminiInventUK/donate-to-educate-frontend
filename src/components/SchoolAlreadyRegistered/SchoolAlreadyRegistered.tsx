import { Link } from 'react-router-dom';
import { HomeOutlined } from '@ant-design/icons';
import styles from './SchoolAlreadyRegistered.module.scss';
import Paths from '@/config/paths';
import { FC } from 'react';

const SchoolAlreadyRegistered: FC = () => {
  return (
    <div className={styles.container}>
      <h2>Your school has already joined Donate to Educate</h2>
      <p>
        For now, one person in each school can join Donate to Educate. In the future, more
        colleagues will be able to join.
      </p>
      <p>
        You may wish to speak with your colleagues to find out who has created your school profile.
      </p>
      <p>
        If you need help,{' '}
        <Link to={Paths.CONTACT} className={styles.link}>
          contact Donate to Educate
        </Link>
        .
      </p>
      <div>
        <HomeOutlined className={styles.homeIcon} /> &nbsp;
        <Link className={styles.link} to={Paths.HOME}>
          Return to homepage
        </Link>
      </div>
    </div>
  );
};
export default SchoolAlreadyRegistered;
