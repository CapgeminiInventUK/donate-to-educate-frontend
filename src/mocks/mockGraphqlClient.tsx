import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { HttpResponse, graphql } from 'msw';
import { ComponentType, FC, ReactNode } from 'react';
import { server } from './server';
import { BrowserRouter as Router } from 'react-router-dom';

const createTestQueryClient = (): QueryClient =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

export const createWrapper = (node: ReactNode): ComponentType => {
  const testQueryClient = createTestQueryClient();
  const MockComponent: FC = () => {
    return (
      <Router>
        <QueryClientProvider client={testQueryClient}>{node}</QueryClientProvider>
      </Router>
    );
  };
  return MockComponent;
};

export const mockApiResponse = (
  queryName: string,
  response: Record<string, unknown> | null,
  throwError = false
): void => {
  server.use(
    graphql.query(queryName, () => {
      // eslint-disable-next-line no-console
      console.log('here');
      if (throwError) {
        return HttpResponse.json({ data: { isError: true } });
      }
      return HttpResponse.json(response);
    })
  );
};
