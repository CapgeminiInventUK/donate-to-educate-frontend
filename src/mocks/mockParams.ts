export const validPostcode = 'LL70 9DJ';
export const invalidPostcode = '123';
export const nonExistentPostcode = 'ZZ1 9ZZ';
