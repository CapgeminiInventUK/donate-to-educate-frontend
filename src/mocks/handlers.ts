import {
  AsyncResponseResolverReturnType,
  GraphQLQuery,
  HttpResponse,
  graphql,
  http,
  passthrough,
} from 'msw';
import getSchoolQueryResponse from './data/getSchoolQuery.json';
import getSchoolsNearbyQueryResponse from './data/getSchoolsNearbyQuery.json';
import { nonExistentPostcode, validPostcode } from './mockParams';

export const handlers = [
  graphql.query('GetSchool', ({ variables }) => {
    if (!String(variables.name).includes('Error')) {
      return HttpResponse.json({
        data: getSchoolQueryResponse,
      });
    }
    return HttpResponse.error() as AsyncResponseResolverReturnType<GraphQLQuery>;
  }),
  graphql.query('GetSchoolsNearby', ({ variables }) => {
    const { postcode } = variables;
    if (postcode === validPostcode) {
      return HttpResponse.json({
        data: getSchoolsNearbyQueryResponse,
      });
    }
    if (postcode === nonExistentPostcode) {
      return HttpResponse.error() as AsyncResponseResolverReturnType<GraphQLQuery>;
    }
  }),
  http.post('https://cognito-identity.eu-west-2.amazonaws.com/', () => {
    return passthrough();
  }),
  http.get('/*.webp', () => {
    return passthrough();
  }),
];
