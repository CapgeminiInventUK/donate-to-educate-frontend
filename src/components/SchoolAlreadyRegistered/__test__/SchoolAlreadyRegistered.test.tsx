import { createWrapper } from '@/mocks/mockGraphqlClient';
import SchoolAlreadyRegistered from '../SchoolAlreadyRegistered';
import { render } from '@testing-library/react';

describe('School already registered', () => {
  it('should render component', () => {
    const Component = createWrapper(<SchoolAlreadyRegistered type="registered" />);
    expect(render(<Component />)).toMatchSnapshot();
  });
});
