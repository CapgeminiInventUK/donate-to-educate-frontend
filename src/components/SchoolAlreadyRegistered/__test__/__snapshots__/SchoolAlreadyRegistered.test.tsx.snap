// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`School already registered > should render component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_container_8cc6ff"
      >
        <h2>
          Your school has already joined Donate to Educate
        </h2>
        <p>
          For now, one person in each school can join Donate to Educate. In the future, more colleagues will be able to join.
        </p>
        <p>
          You may wish to speak with your colleagues to find out who has created your school profile.
        </p>
        <p>
          If you need help,
           
          <a
            class="_link_8cc6ff"
            href="/contact"
          >
            contact Donate to Educate
          </a>
          .
        </p>
        <div>
          <span
            aria-label="home"
            class="anticon anticon-home _homeIcon_8cc6ff"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="home"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
              />
            </svg>
          </span>
            
          <a
            class="_link_8cc6ff"
            href="/"
          >
            Return to homepage
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_container_8cc6ff"
    >
      <h2>
        Your school has already joined Donate to Educate
      </h2>
      <p>
        For now, one person in each school can join Donate to Educate. In the future, more colleagues will be able to join.
      </p>
      <p>
        You may wish to speak with your colleagues to find out who has created your school profile.
      </p>
      <p>
        If you need help,
         
        <a
          class="_link_8cc6ff"
          href="/contact"
        >
          contact Donate to Educate
        </a>
        .
      </p>
      <div>
        <span
          aria-label="home"
          class="anticon anticon-home _homeIcon_8cc6ff"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="home"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
            />
          </svg>
        </span>
          
        <a
          class="_link_8cc6ff"
          href="/"
        >
          Return to homepage
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
